src/components/application_manager/test/reset_timeout_handler_test.cc

```diff
@@ -0,0 +1,175 @@
+      .WillByDefault(ReturnRef(kDefaultTimeout));
+
+  EXPECT_CALL(app_mngr_, GetResetTimeoutHandler())
+      .WillOnce(ReturnRef(*reset_timeout_handler_));
+  EXPECT_CALL(app_mngr_, updateRequestTimeout(_, _, kTimeout));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+  reset_timeout_handler_->on_event(event);
+}
+
+TEST_F(ResetTimeoutHandlerTest, on_event_OnResetTimeout_missed_reset_period) {
+  std::shared_ptr<SubscribeWayPointsRequest> command =
+      CreateCommand<SubscribeWayPointsRequest>();
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(app_mngr_, IsAppSubscribedForWayPoints(_))
+      .WillByDefault(Return(false));
+  ON_CALL(app_mngr_, IsAnyAppSubscribedForWayPoints())
+      .WillByDefault(Return(false));
+  ON_CALL(app_mngr_, GetNextHMICorrelationID())
+      .WillByDefault(Return(kRequestId));
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).WillOnce(Return(true));
+
+  Event event(hmi_apis::FunctionID::BasicCommunication_OnResetTimeout);
+  smart_objects::SmartObject notification_msg;
+  notification_msg[msg_params][request_id] = kRequestId;
+  event.set_smart_object(notification_msg);
+
+  ON_CALL(app_mngr_, get_settings())
+      .WillByDefault(ReturnRef(app_mngr_settings_));
+  ON_CALL(app_mngr_settings_, default_timeout())
+      .WillByDefault(ReturnRef(kDefaultTimeout));
+
+  EXPECT_CALL(app_mngr_, GetResetTimeoutHandler())
+      .WillOnce(ReturnRef(*reset_timeout_handler_));
+  EXPECT_CALL(app_mngr_, updateRequestTimeout(_, _, kDefaultTimeout));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+  reset_timeout_handler_->on_event(event);
+}
+
+TEST_F(ResetTimeoutHandlerTest, on_event_OnResetTimeout_invalid_request_id) {
+  std::shared_ptr<SubscribeWayPointsRequest> command =
+      CreateCommand<SubscribeWayPointsRequest>();
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(app_mngr_, IsAppSubscribedForWayPoints(_))
+      .WillByDefault(Return(false));
+  ON_CALL(app_mngr_, IsAnyAppSubscribedForWayPoints())
+      .WillByDefault(Return(false));
+  ON_CALL(app_mngr_, GetNextHMICorrelationID())
+      .WillByDefault(Return(kRequestId));
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).WillOnce(Return(true));
+
+  Event event(hmi_apis::FunctionID::BasicCommunication_OnResetTimeout);
+  smart_objects::SmartObject notification_msg;
+  notification_msg[msg_params][reset_period] = kTimeout;
+  notification_msg[msg_params][request_id] = 0u;
+  event.set_smart_object(notification_msg);
+
+  ON_CALL(app_mngr_, get_settings())
+      .WillByDefault(ReturnRef(app_mngr_settings_));
+  ON_CALL(app_mngr_settings_, default_timeout())
+      .WillByDefault(ReturnRef(kDefaultTimeout));
+
+  EXPECT_CALL(app_mngr_, GetResetTimeoutHandler())
+      .WillOnce(ReturnRef(*reset_timeout_handler_));
+  EXPECT_CALL(app_mngr_, updateRequestTimeout(_, _, kTimeout)).Times(0);
```

Comment:
@LitvinenkoIra right here we can miss the kTimeout value for expectation

Reply:
@LuxoftAKutsan fixed 7d67c458fed62cd11e953954fa6285e19842d46c